File: src/main/java/ru/AlexeyFedechkin/znatoki/StudyRussianBot/Object/Enums/UserStatus.java
/*
 * Author: Fedechkin Alexey Borisovich
 * last modified: 22.06.19 12:55
 * Copyright (c) 2019
 */

package ru.AlexeyFedechkin.znatoki.StudyRussianBot.Object.Enums;

/**
 *
 */
public enum UserStatus {
    NONE,
    TESTING, // passing the rules
    WAIT_COUNT_OF_WORD, // input count of word for testing
    WAIT_KEY // input activated key
}

File: src/main/java/ru/centralhardware/znatoki/studyRussianBot/objects/enums/UserStatus.kt
package ru.centralhardware.znatoki.studyRussianBot.objects.enums

/**
 * user status
 * bot used state machine for control state of user
 * Copyright © 2019-2021 Fedechkin Alexey Borisovich. Contacts: alex@centralhardware.ru
 */
enum class UserStatus {
    /**
     *user have not any acton or user just start bot
     */
    NONE,
    /**
     * passing the rules
     */
    TESTING,
    /**
     * input count of word for testing
     */
    WAIT_COUNT_OF_WORD,
    /**
     * input activated key
     */
    WAIT_KEY,
}